util/reggen/top_uvm_reg.sv.tpl

```diff
@@ -35,9 +35,10 @@
       if_suffix = '' if if_name is None else '_' + if_name
       esc_if_name = block.name.lower() + if_suffix
       if_desc = '' if if_name is None else '; interface {}'.format(if_name)
-%>\
+      hier_path = '' if block.hier_path is None else block.hier_path + "."
+ %>\
 // Block: ${block.name.lower()}${if_desc}
-${make_ral_pkg(dv_base_prefix, top.regwidth, '', rb, esc_if_name)}
+${make_ral_pkg(dv_base_prefix, top.regwidth, hier_path, rb, esc_if_name)}
```

Comment:
Hmm, that's interesting: when I added the `''` here, I thought I was reproducing what was in the code before. If this change is correct, please could you remove this argument completely from `make_ral_pkg` and move the `hier_path` calculation into `uvm_reg_base.sv.tpl`?

Reply:
@rswarbrick thanks for looking at this.

currently `hier_path` is calculated outside the `make_ral_pkg`. We don't use `block` in `make_ral_pkg`, do you want to do that?
https://github.com/lowRISC/opentitan/blob/4ca4a2f07bce8912d6adc10a56584319cea67491/util/reggen/uvm_reg.sv.tpl#L15-L17

Reply:
Ah, good point: I hadn't realised. Yeah, I think it makes sense to pass the block in (which also lets you get rid of the `regwidth` argument). I did it like this originally because I was disentangling the top-level bits (which don't have a block) from the rest: looks like I should have looked again before pushing!

Reply:
@rswarbrick 
as you said, regwidth can be from `Top`, like this one. maybe it's ok to pass regwidth and hier_path since we only need these 2 attributes from block for now. WDYT?
https://github.com/lowRISC/opentitan/blob/41d815de155cb672b069155963395d104b1d71a6/util/reggen/top_uvm_reg.sv.tpl#L137

Reply:
Aargh, I'd forgotten you had this patch pending and my changes in #5856 will clash, sorry. If you can bear to (and assuming that my changes look ok), would you mind rebasing on top of them? I think the refactoring you're describing here is probably still the right thing to do.